doc/specs/#5000 - Process Model 2.0/#5000 - Process Model 2.0.md

```diff
@@ -0,0 +1,1204 @@
+"Pope Stephen II" scenario). In this scenario, if at any point a peasant
+notices that the monarch has died, the peasant will need to begin checking who
+the new monarch is again.
+
+There is certain to be a long tail of edge cases we'll discover as a product of
+this change. We'll need to be prepared to see an inevitable decrease in initial
+reliability numbers, and increased bug reports for a time. Additionally, these
+bugs will likely be fairly difficult to track down. Initially, it may help us to
+track these bugs down if we add additional tracing around each of the
+inter-process calls we might make. Some example situations might include:
+* When a process creates a new content process
+* When a content process is attached by a window
+* When a monarch is elected
+* when a peasant receives its new ID from a monarch
+* when either a window or content process _safely_ exits
+
+In any and all of these situations, we'll want to try and be as verbose as
+possible in the logging, to try and make tracking which process had the error
+occur easier.
+
+In the long run, this may end up _increasing_ reliability, as crashes in the
+buffer should no longer cause the _entire_ terminal application to crash.
+
+
+</td>
+</tr>
+<tr>
+<td><strong>Compatibility</strong></td>
+<td>
+
+The biggest concern regarding compatibility will be ensuring that the Universal
+app version of the Windows Terminal will still work as before. Although that
+application isn't used by any customer-facing scenarios currently, it still
+represents some long term goals for the Terminal.  We'll probably need to
+disable tear-out _entirely_ on the universal app, at least to begin with.
+Additionally, we'll need to ensure that all the controls in the universal
+application are in-proc controls, not using the window/content split process
+model.
+
+I'm also concerned that each individual atomic step along the path towards this
+goal still works. For more on this topic, see [Implementation
+Plan](#implementation-plan).
+
+</td>
+</tr>
+<tr>
+<td><strong>Performance, Power, and Efficiency</strong></td>
+<td>
+
+There's no dramatic change expected here. There may be a minor delay in the
+spawning of new terminal instances, due to requiring cross-process hops for the
+instantiation of the content process.
+
+Additionally, minor delays are expected to be introduced during startup for the
+initial setup of the monarch/peasant relationship.
+
+</td>
+</tr>
+</table>
+
+## Potential Issues
+
+
+### Extensions & non-terminal content
+
+We've now created a _very_ terminal-specific IPC mechanism for transferring
+_terminal_ state from one thread to another. However, what happens when we start
+to have panes that contain non-terminal content in them? This is a scenario
+that's most often associated with the concept of extensions in the Terminal, but
+perhaps more relevantly could affect the Settings UI.
+
+The Settings UI is something we intend on shipping in the Terminal as a part of

```

Comment:
(This is me rambling, but I figured I'd jot this down somewhere)

Settings UI ideas:
- maintaining only one Settings UI tab open at a time
  - today, we just do a "switchToTab" command. When multiple windows are open, we could do a two step approach. (1) Monarch keeps track of the window (if one exists) that has SUI open. (2) navigate to that window, then invoke the "switchToTab" command.
- tab tearout
  - Most of the meta-data for the Settings UI is really just the clone of `CascadiaSettings`. That should be fairly easy to hand off. The only other thing we would need is the `tag` and relevant `NavigationState` object. That way, we ensure we're on the same page.